intelmq/bots/parsers/shadowserver/_config.py

```diff
@@ -204,6 +205,18 @@ def convert_date_utc(value: str) -> Optional[str]:
     return harmonization.DateTime.sanitize(value + '+00:00')
 
 
+_base64_alphabet = set('ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=')
+
+
+def maybe_base64(value: Optional[str]) -> Optional[str]:
+    if not value:
+        return None
+    elif set(value).issubset(_base64_alphabet):
+        return value
+
+    return base64.b64encode(value.encode()).decode()
```

Comment:
Do you want to *decode from* base64 or *encode to* base64? Currently the code does encode, but that doesn't really make sense for me. If you want to decode, the latter should work:

```suggestion
def decode_base64(value: Optional[str]) -> Optional[str]:
    """
    Lazy decoding of base64-encoded data.
    If the character set indicates a base64-encoded content, decoding is attempted and the result returned.
    Otherwise, or if the decoding fails, the original string is returned.
    """
    if not value:
        return None
    elif set(value).issubset(_base64_alphabet):
        try:
            return base64.b64encode(value.encode()).decode()
        except binascii.Error:
            return Value
    else:
        return value
```

Reply:
Err. It's meant to be encode, actually, so as to keep the data consistently base64-encoded, whereas currently it's sometimes base64-encoded, sometimes not. Decoding arbitrary binary values from POST queries as UTF8 woudln't work very well.

Reply:
Ah, okay. Then I propose to chose a different function name, plus docstring and don't use the issubset for checking if the value is base64, but a try-except-block decoding the value.